src/hotspot/share/gc/shenandoah/shenandoahFreeSet.cpp

```diff
@@ -305,47 +903,25 @@ HeapWord* ShenandoahFreeSet::allocate_contiguous(ShenandoahAllocRequest& req) {
       used_words = ShenandoahHeapRegion::region_size_words();
     }
 
+    r->set_update_watermark(r->bottom());
```

Comment:
Why is that here? It means we would update-refs in this humongous object, but why is it needed in a new object?

Reply:
Good catch.  I suspect this code was accidentally introduced by GenShen engineers before we had a good grasp of the update_watermark role.  I'm removing this.